src/components/Fund/BudgetAddModal/BudgetAddModal.js

```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+
+import { useShowToast } from '@folio/stripes-acq-components';
+import { stripesConnect } from '@folio/stripes/core';
+
+import BudgetAddModalForm from './BudgetAddModalForm';
+import {
+  budgetResource,
+  fiscalYearsResource
+} from '../../../common/resources';
+import getFiscalYearsForSelect from '../../../Utils/getFiscalYearsForSelect';
+import { BUDGET_STATUSES } from '../../Budget/constants';
+
+
+const BudgetAddModal = ({ history, mutator, resources, onClose, fund, budgetStatus }) => {
+  const showCallout = useShowToast();
+  const fiscalYears = getFiscalYearsForSelect(resources);
+
+  const createBudget = useCallback(
+    async (formValue) => {
+      try {
+        const fiscalYear = fiscalYears.find(year => year.value === formValue.fiscalYearId);
+        const budget = await mutator.budget.POST({
+          ...formValue,
+          fundId: fund.id,
+          name: `${fund.code}-${fiscalYear.code}`,
+        });
+        const { name, id } = budget;
+        showCallout('ui-finance.budget.hasBeenCreated', 'success', { name, fundName: fund.name });
+        const path = `/finance/budget/${id}/view`;
+
+        history.push(path);
+      } catch (e) {
+        showCallout('ui-finance.budget.hasNotBeenCreated', 'error');
+      }
+    },
+    [history, fiscalYears, fund, mutator]
```

Comment:
`fiscalYears` is always new array

Reply:
I tried memoized `fiscalYears`. Take a look my changes.